### ISSUE TITLE
Api Gateway OPTIONS returns 404 on valid api endpoint.

### ISSUE THREAD

**Issue by vangorra** [localstack/localstack#449]
Couldn't reopen #399 as I didn't close it.

Just tested this now. I'm getting a 404 when calling OPTIONS on the api gateway url. I can run curl requests against my endpoint correctly, only OPTIONS method returns a 404.
```
user@host:/code$ curl -I -X OPTIONS http://localstack-core:4567/restapis/5A-Z18A-Z07A-Z22/local/_user_request_/serviceName
HTTP/1.1 404 Not Found
Server: BaseHTTP/0.3 Python/2.7.13
Date: Fri, 03 Nov 2017 17:06:33 GMT
Content-Length: 95
Access-Control-Allow-Origin: *
Access-Control-Allow-Methods: HEAD,GET,PUT,POST,DELETE,OPTIONS,PATCH
Access-Control-Allow-Headers: authorization,content-type,content-md5,x-amz-content-sha256,x-amz-date,x-amz-security-token,x-amz-user-agentReferrer Policy:no-referrer-when-downgrade
```

**Comment by whummer** [comment]
Fixed in #814 - closing.

**Comment by vinicri** [comment]
I'm getting this issue with this response body
{"message": "API Gateway integration type \"MOCK\" for method \"OPTIONS\" not yet implemented"}
